File: posts/mathworks-interview/500.py
class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        charRowMapping = {} # 搞一个字符到排号的映射
        charRowMapping.update({v: 0 for v in "qwertyuiop"}) # 第一排的所有字符
        charRowMapping.update({v: 1 for v in "asdfghjkl"}) # 第二排的所有字符
        charRowMapping.update({v: 2 for v in "zxcvbnm"}) # 第三排的所有字符
        res = []

        for word in words:
            if len(set(charRowMapping[v] for v in word.lower())) == 1: # 如果单词里面所有的字符都在同一排
                res.append(word)

        return res

File: posts/mathworks-interview/500.rs
struct Solution;

use std::collections::HashMap;
use std::collections::HashSet;

impl Solution {
    pub fn find_words(words: Vec<String>) -> Vec<String> {
        let mut charRowMapping: HashMap<char, i32> = HashMap::new();
        charRowMapping.extend("qwertyuiop".chars().map(|v| (v, 0)));
        charRowMapping.extend("asdfghjkl".chars().map(|v| (v, 1)));
        charRowMapping.extend("zxcvbnm".chars().map(|v| (v, 2)));

        let mut res: Vec<String> = vec![];

        for word in words.iter() {
            let rows: HashSet<i32> = word.to_lowercase().chars().map(|v| *charRowMapping.get(&v).unwrap()).collect();
            if rows.len() == 1 {
                res.push(word.clone());
            }
        }

        return res;
    }
}